### PR TITLE
bigint.remove() renamed to bigint.removeFactor()

### LINKED ISSUE #17732: [Library Stabilization] Should we rename BigInteger's `bigint.remove()` method?
This method name does not really seem clear to me.  The function removes all instances of the factor (`f`) arg from the other arg (`a`), storing the result in `this` and returning the number of times the factor was removed.  I think "remove" conflicts pretty regularly with other remove methods.  Maybe `bigint.removeFac()` would be better?  I'm open to other suggestions as well

This function is defined here: https://gmplib.org/manual/Number-Theoretic-Functions

### PR DESCRIPTION
**Pull request by tiger-yash**
This PR closes #17732

### Description
* Changed the function name
* Added the required Deprecation Warning test.

### CODE REVIEW

File: modules/standard/BigInteger.chpl [hunk 1]
```diff
@@ -3153,7 +3153,18 @@ When ``n/d`` does not produce an integer, this method may produce incorrect resu
...
+
+       bigint.remove is deprecated, use bigint.removeFactor instead
+  */
+  deprecated
+  "bigint.remove is deprecated, use bigint.removeFactor instead"
   proc bigint.remove(const ref a: bigint, const ref f: bigint) : uint {
+    return this.removeFactor(a,f);
+  }
+
+  proc bigint.removeFactor(const ref a: bigint, const ref f: bigint) : uint {
```
Comment (lydia-duncan) [inline modules/standard/BigInteger.chpl:3167]: I might be tempted to rename these arguments to `x` and `fac` respectively while you're here, unless you have better name suggestions.

**Review by lydia-duncan — COMMENTED** [review]
I'm expecting there to be a test that uses this method which will fail and need to be updated to use the new method.  It'll likely turn up when I run full testing over the branch but if you want to find it faster, it's probably in `test/library/standard/GMP` (I'd just run that directory locally and see what fails).

Other than, I'll just wait to see how you adjust the argument names before running testing

**Comment by tiger-yash** [comment]
@lydia-duncan I have added a Helper function for the failing case and created a test as well.
For the same condition I decided to return 0 and set `this=0` as these have to be integer types, although `undefined` seems much more appropriate.

File: modules/standard/BigInteger.chpl [hunk 2]
```diff
@@ -3163,30 +3163,54 @@ When ``n/d`` does not produce an integer, this method may produce incorrect resu
...
     return this.removeFactor(a,f);
   }
+  /*
+    .. warning::
 
+       a and f are deprecated - please use x and fac respectively
+  */
+  pragma "last resort"
+  deprecated
+  "a and f are deprecated - please use x and fac respectively"
```
Comment (lydia-duncan) [inline modules/standard/BigInteger.chpl:3173]: removeFactor is a new function for your PR, you don't need to have a deprecation warning when changing the argument names :)

Comment (tiger-yash) [reply modules/standard/BigInteger.chpl:3173]: sure, I'll correct it.

File: modules/standard/BigInteger.chpl [hunk 3]
```diff
@@ -3163,30 +3163,54 @@ When ``n/d`` does not produce an integer, this method may produce incorrect resu
...
   proc bigint.removeFactor(const ref a: bigint, const ref f: bigint) : uint {
-    var ret: c_ulong;
+    return this.removeFactor(x=a,fac=f);
+  }
+  // This helper is intended for use only when the factor is 0
+  // Division by 0 is undefined and it results in a
+  // Floating point exception error.
+  pragma "no doc"
+  proc bigint.removeFactorZeroHelper(const ref x: bigint, const ref fac: bigint) : uint {
```
[the remark this answers is not in the document]
Comment (tiger-yash) [reply modules/standard/BigInteger.chpl:3171]: Thanks @lydia-duncan, I hope the tests look fine.

**Comment by lydia-duncan** [comment]
Just one failure with testing:
```
[Error matching program output for library/standard/GMP/ldelaney/gmp_Bigint_number_theoretic]
```
Looks like there's a use of remove that should be updated to removeFactor

**Review by lydia-duncan — APPROVED** [review]

**Pull request merged**